component/src/test/java/org/ballerinalang/kafka/test/ssl/KafkaConsumerAndProducerWithSSLTest.java

```diff
@@ -0,0 +1,173 @@
+import org.ballerinalang.model.values.BValue;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import static org.ballerinalang.kafka.test.utils.Constants.KAFKA_BROKER_PORT;
+import static org.ballerinalang.kafka.test.utils.Constants.ZOOKEEPER_PORT_1;
+
+/**
+ * Test cases for ballerina.kafka consumer and producer with SSL.
+ */
+@Test(singleThreaded = true)
+public class KafkaConsumerAndProducerWithSSLTest {
+    private CompileResult result;
+    private static File dataDir;
+    private static KafkaCluster kafkaCluster;
+    private Path resourceDir = Paths.get("src/test/resources").toAbsolutePath();
+    private String configFile = "ssl/kafka_ssl.bal";
+
+    @BeforeClass
+    public void setup() throws IOException {
+        Properties prop = new Properties();
+        prop.put("listeners", "SSL://localhost:9094");
+        prop.put("security.inter.broker.protocol", "SSL");
+        prop.put("ssl.client.auth", "required");
+        prop.put("ssl.keystore.location", resourceDir + "/keystores-truststores/kafka.server.keystore.jks");
+        prop.put("ssl.keystore.password", "test1234");
+        prop.put("ssl.key.password", "test1234");
+        prop.put("ssl.truststore.location", resourceDir + "/keystores-truststores/kafka.server.truststore.jks");
+        prop.put("ssl.truststore.password", "test1234");
+        kafkaCluster = kafkaCluster().deleteDataPriorToStartup(true)
+                .deleteDataUponShutdown(true).withKafkaConfiguration(prop).addBrokers(1).startup();
+        kafkaCluster.createTopic("test-topic-ssl", 2, 1);
+        //Setting the keystore and trust-store file paths
+        setFilePath(resourceDir.toString() + "/" + configFile, "<FILE-PATH>", resourceDir.toString()
+                + "/keystores-truststores");
+        result = BCompileUtil.compile(configFile);
+    }
+
+    @Test(description = "Test SSL produce")
+    public void testKafkaProducerWithSSL() {
+        BValue[] returnBValues;
+        int msgCount = 0;
+        while (true) {
+            returnBValues = BRunUtil.invoke(result, "funcTestKafkaProduceWithSSL");
+            msgCount++;
+            if (msgCount == 10) {
+                break;
+            }
+            Assert.assertEquals(returnBValues.length, 1);
+        }
+    }
+
+    @Test(description = "Test consumer polling with SSL", dependsOnMethods = {"testKafkaProducerWithSSL"})
+    @SuppressWarnings("unchecked")
+    public void testKafkaConsumerWithSSL() {
+        BValue[] returnBValues;
+        int msgCount = 0;
+        while (true) {
```

Comment:
Shall we add `awaitality` function here ? Otherwise, if the test fails, this loop may never exit.